### PR TITLE
chore(dep): add autosuggest-highlight dependency

### PR DESCRIPTION
**Pull request by gkartalis**
The type of this PR is: **Enhancement**

This PR resolves []

### Description

More info can be found on this [RFC](https://github.com/artsy/eigen/issues/5564)

# Screenshots

| before | before bug display with comma | after | after display with comma |
|---|---|---|---|
|Simulator Screen Shot - iPhone 8 - 2021-10-01 at 10 07 16|Simulator Screen Shot - iPhone 8 - 2021-10-01 at 10 04 40|Simulator Screen Shot - iPhone 8 - 2021-10-01 at 10 04 29|Simulator Screen Shot - iPhone 8 - 2021-10-01 at 10 07 21|

bonus screenshot of new implementation that matches results without user typing diacritics of letter
simulator_screenshot_0328FF9F-F95A-47E6-B287-E87416564673

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- changed the logic behind highlighting matching queries on search - gkartalis

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (changed the logic behind highlighting matching queries on search - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c6c0063c68ac64b42140a4ebd17ee91bfc026921
</p>

**Review by pvinis — APPROVED** [review]
very nice! Could you add a screenshot of wnere this is shown?

**Comment by gkartalis** [comment]
cc @anastasiapyzhik @dimatretyak

**Comment by dimatretyak** [comment]
Amazing 🤩

File: src/lib/Scenes/Search/ResultWithHighlight.tsx [hunk 1]
```diff
@@ -1,68 +1,24 @@
...
-      </SmallText>
-      {nonMatch2}
-    </SmallText>
+    <Text variant="xs" numberOfLines={1}>
+      {parts.map((part, i) => (
+        <Text
+          key={i}
+          variant="xs"
+          color={part.highlight ? "blue100" : "black100"}
+          weight={part.highlight ? "medium" : "regular"}
```
Comment (dimatretyak) [inline src/lib/Scenes/Search/ResultWithHighlight.tsx:17]: 👏

**Review by dimatretyak — APPROVED** [review]

**Review by anastasiapyzhik — APPROVED** [review]

**Pull request merged**